### PR TITLE
Services: PTM::Added CheckNew3DS and ATP::Uknown_0x01010000 to help with #840

### PR DESCRIPTION
**Pull request by Zaneo**
Added stubbed ATP::Uknown_0x01010000
Added PTM::CheckNew3Ds

### CODE REVIEW

File: src/core/hle/service/apt/apt.cpp [hunk 1]
```diff
@@ -327,6 +329,23 @@ void GetAppCpuTimeLimit(Service::Interface* self) {
...
 }
 
+void Unknown_0x01010000(Service::Interface* self){
+    u32* cmd_buff = Kernel::GetCommandBuffer();
+
+    cmd_buff[1] = RESULT_SUCCESS.raw;
+
+    // When a certain NS state field (internal NS state in .data/.bss.) is non-zero, the output value is zero
+    // If/Else is to indicate that a check should occur here.
+    if (false){
```
Comment (lioncash) [inline src/core/hle/service/apt/apt.cpp]: tbh I'm not really for half-implementing service functions like this.

File: src/core/hle/service/apt/apt.h [hunk 2]
```diff
@@ -299,6 +299,15 @@ void PrepareToStartLibraryApplet(Service::Interface* self);
  */
 void StartLibraryApplet(Service::Interface* self);
 
+/**
+* From 3Ds Brew:
+*	This writes an output u8 to cmdreply indexword[2].
+*	This uses PTMSYSM:CheckNew3DS.
+*	When a certain NS state field (internal NS state in .data/.bss.) is non-zero, the output value is zero, otherwise the output is from PTMSYSM:CheckNew3DS. 
+*	Normally this NS state field is zero, however this state field is set to 1 when APT:PrepareToStartApplication is used with flags bit8 is set.
```
Comment (Zaneo) [inline src/core/hle/service/apt/apt.h]: The pre-commit hook may have missed the trailing whitespcae?

File: src/core/hle/service/apt/apt.cpp (hunk 1 above)
Comment (Zaneo) [inline src/core/hle/service/apt/apt.cpp]: This was mostly to help move #840 along. It looked like this allowed that to progress further. I don't have any hardware to track down this state field, which is apparently title specific.

**Comment by bunnei** [comment]
I'm curious if making this function say Citra is a N3DS helps ZMM along... I guess I'll test that later.

**Comment by Subv** [comment]
I'm against this, stubbing these functions like this brings nothing useful. The real behavior should be investigated

**Comment by neobrain** [comment]
Seconding @Subv's comment, especially since the PR description doesn't even describe why this change was deemed useful.

Also, it's "3DBrew", not "3Ds Brew".

**Comment by Zaneo** [comment]
Not disagreeing, in this case is there much risk?

"When a certain NS state field is non-zero, the output value is zero, otherwise the output is from PTMSYSM:CheckNew3DS."

Currently Old3DS (Value = 0) is returned always, which means that the condition is always satisfied.

Maybe a comment to say that until the NS statefield is discovered it should return 0 instead of CheckNew3DS?

**Comment by Subv** [comment]
I also don't like how this introduces yet another "CheckNew3DS" function, we ought to have a centralized point for all of this (see the CFG functions that also query the 3DS type)

**Comment by Zaneo** [comment]
No rush.

**Comment by yuriks** [comment]
For me it would be a safe bet to say that this "internal field" that's being talked about is some sort of flag that says that the process is a Old3DS compatibility-mode process. I think we can just implement this as returning false until we investigate more comprehensive n3DS support.

**Comment by purpasmart96** [comment]
Plus all the formatting was all wrong.

**Pull request closed without merging**